cmd/kubernetes.go

```diff
@@ -0,0 +1,818 @@
+		if err != nil {
+			warn("failed to write log to file: %s", err)
+		}
+		err = ioutil.WriteFile(config.ManagerDir+"/kubeSchedulerLog", []byte(kubeSchedulerLog), 0644)
+		if err != nil {
+			warn("failed to write kubeSchedulerLog to file: %s", err)
+		}
+		err = ioutil.WriteFile(config.ManagerDir+"/kubeSchedulerControllerLog", []byte(kubeSchedulerControllerLog), 0644)
+		if err != nil {
+			warn("failed to write kubeSchedulerControllerLog to file: %s", err)
+		}
+
+		// teardown kubernetes resources we created
+		if len(kubeNamespace) != 0 {
+			info("deleting namespace %s", kubeNamespace)
+			err = client.TearDown(kubeNamespace)
+			if err != nil {
+				die("failed to delete the kubernetes resources previously created: %s", err)
+			}
+		} else {
+			info("deleting namespace %s", resources.Details["namespace"])
+			err = client.TearDown(resources.Details["namespace"])
+			if err != nil {
+				die("failed to delete the kubernetes resources previously created: %s", err)
+			}
+		}
+
+		err = os.Remove(filepath.Join(config.ManagerDir, "kubernetes_resources"))
+		if err != nil {
+			warn("failed to delete the kubernetes resources file: %s", err)
+		}
+		err = os.Remove(filepath.Join(config.ManagerDir + "/key.pem"))
+		if err != nil {
+			warn("failed to delete key.pem: %s", err)
+		}
+		err = os.Remove(filepath.Join(config.ManagerDir + "/cert.pem"))
+		if err != nil {
+			warn("failed to delete cert.pem: %s", err)
+		}
+		err = os.Remove(filepath.Join(config.ManagerDir + "/ca.pem"))
+		if err != nil {
+			warn("failed to delete ca.pem: %s", err)
+		}
+		err = os.Remove(filepath.Join(config.ManagerDir + "/client.token"))
+		if err != nil {
+			warn("failed to delete the client token : %s", err)
+		}
+
+		info("deleted all kubernetes resources previously created")
+
+		// kill the port forwarders
+		if fmRunning {
+			err = killProcess(fmPid)
+			if err == nil {
+				err = os.Remove(fmPidFile)
+				if err != nil && !os.IsNotExist(err) {
+					warn("failed to remove the forwarder pid file %s: %s", fmPidFile, err)
+				}
+			}
+		}
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(kubeCmd)
+	kubeCmd.AddCommand(kubeDeployCmd)
+	kubeCmd.AddCommand(kubeTearDownCmd)
+
+	// flags specific to these sub-commands
+	defaultConfig := internal.DefaultConfig(appLogger)
+	kubeDeployCmd.Flags().StringVarP(&podPostCreationScript, "script", "s", defaultConfig.CloudScript, "path to a start-up script that will be run on each pod created")
+	kubeDeployCmd.Flags().IntVarP(&serverKeepAlive, "keepalive", "k", defaultConfig.CloudKeepAlive, "how long in seconds to keep idle spawned servers alive for; 0 means forever")
+	kubeDeployCmd.Flags().IntVarP(&maxServers, "max_servers", "m", defaultConfig.CloudServers+1, "maximum number of servers to spawn; 0 means unlimited (default 0)")
+	kubeDeployCmd.Flags().StringVar(&podDNS, "network_dns", defaultConfig.CloudDNS, "comma separated DNS name server IPs to on the created pods")
```

Comment:
The description of this option doesn't seem to make sense

Reply:
I'm unsure of the specific flag you mean, or do you mean in general?

Reply:
The last one here; I don't know what "comma separated DNS name server IPs to on the created pods" means.

Reply:
I've just remembered I meant to ask you, currently it does nothing, however, It may be useful??

https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-config

Is the sort of idea, pods can be given individual DNS configurations. It may be useful, but I'm inclined to remove it until requested?